.github/workflows/lint.yml

```diff
@@ -0,0 +1,34 @@
+name: Lint
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
+        
+    - name: Install Ruff
+      run: pip install ruff
+      
+    - name: Lint with Ruff
+      run: |
+        ruff check .
+        ruff fix .
+        
+    - name: Commit changes
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: 'Automatically fix Ruff issues'
+        branch: ${{ github.head_ref }}
```

Comment:
Hi @saalUW! Thanks for working on this. I think the linting is probably good to be done in a pre-commit rather than a post through workflow. To be honest I am not very familiar with [pre-commit](https://github.com/pre-commit/pre-commit), but probably good to explore.